packages/docgen/lib/get-type-annotation.js

```diff
@@ -531,6 +558,9 @@ module.exports =
 			case 'return': {
 				return getReturnTypeAnnotation( token );
 			}
+			case 'type': {
```

Comment:
Just to clarify what these cases mean -- we currently support parsing the types out of `@param` and `@return` in doc comments? And this adds support for `@type`?

Reply:
> we currently support parsing the types out of @param and @return in doc comments? And this adds support for @type?

Not quite. We don't parse the types from `@param` or `@return` when a type is present in the JSDoc annotation. In those cases it's caught by the `if ( tag.type )` at the start of this function, we just use the type that is written in there.

What we're doing here is adding support for type extraction from the TypeScript. "type" in this case is just a dummy tag we've generated to tell `getTypeAnnotation` that we want to get the type annotation of a static variable.